common/model-views.cpp

```diff
@@ -2864,17 +2864,23 @@ namespace rs2
         texture_buffer* texture_frame = nullptr;
         points p;
         frame f{}, res{};
+
+        std::map<int, frame> last_frames;
         try
         {
             while (ppf.resulting_queue.poll_for_frame(&f))
```

Comment:
Can we drop the "while" loop in favor of "for" to remove the potential for other bug of the same nature ?